grim.go

```diff
@@ -248,3 +250,21 @@ Or to use the server defaults, remove the entry %q`
 func buildTruncatedMessage(truncateID string) string {
 	return fmt.Sprintf(truncatedMessage, truncateID, truncateID, truncateID)
 }
+
+func shouldSkip(hook *hookEvent) *string {
+	var message *string
+	switch {
+	case hook.Deleted:
+		message = getStringPtr(fmt.Sprintf("because it was on a deleted branch"))
+	case hook.EventName == "push":
+	case hook.Action == "opened":
+	case hook.Action == "reopened":
+	case hook.Action == "synchronize":
+		message = nil
```

Comment:
Case statements don't automatically [fallthrough](https://golang.org/ref/spec#Fallthrough_statements) in Go.